sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java

```diff
@@ -79,15 +75,6 @@ public KeyVaultOperation(final SecretClient secretClient,
         final long refreshInMillis = refreshDuration.toMillis();
         if (refreshInMillis > 0) {
             synchronized (KeyVaultOperation.class) {
-                if (timer != null) {
-                    try {
-                        timer.cancel();
-                        timer.purge();
-                    } catch (RuntimeException runtimeException) {
-                        LOGGER.error("Error of terminating Timer", runtimeException);
-                    }
-                }
-                timer = new Timer(true);
```

Comment:
Doing like this will make the problem fixed in https://github.com/Azure/azure-sdk-for-java/pull/16143 appear again.